apis/apiextensions/v1/composition_types.go

```diff
@@ -205,6 +210,7 @@ const (
 	PatchTypeFromCompositeFieldPath PatchType = "FromCompositeFieldPath" // Default
 	PatchTypePatchSet               PatchType = "PatchSet"
 	PatchTypeToCompositeFieldPath   PatchType = "ToCompositeFieldPath"
+	PatchTypeFromConstantValue      PatchType = "FromConstantValue"
```

Comment:
Currently I believe we support patching from a constant value to a composed resource. I wonder if there's any value / use cases in patching from a constant value back to the composite resource. Similar to how we have `FromCompositeFieldPath` and `ToCompositeFieldPath`.

It seems like it would be fairly straightforward to support this in either "direction" (we'd just need to additionally support calling `applyFromConstantValuePatch(from)`). The tricky part is figuring out what we'd call the two patch types.

My strawman proposal would be:

* `CompositeFieldPathFromComposedValue` patches the composite resource.
* `ComposedFieldPathFromConstantValue` patches the composed resource.

I'd love to hear better options though, because these strings are a little too long for my liking and we've previously heard that it's hard to distinguish between `Composite` and `Composed` in strings like this.

Reply:
Hrm, actually, on reflection I don't think it really makes sense to patch from a constant value back to the composite resource - at least not in this particular context. 🤔 We're patching from an array of patches that are associated with a specific composed resource, but I can't imagine that a constant value will really be associated with a particular composed resource.

I wonder if it makes sense to support an array of patches from the Composition in general back to the XR? In some cases we can just use OpenAPI defaulting at the XR level (e.g. if there should be a constant default value regardless of Composition) but I could imagine wanting to set a constant XR value based on which Composition was selected.

Note - we don't need to bring this idea into scope for this PR. What I'd like to determine is whether we really think we'd need to patch from a constant back to the XR _from a particular composed resource's templates array_. If we don't think this is a real use case we can just ignore this thread and open a separate issue to track patching from a constant value to the XR.